<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Übung Objektorientierte Programmierung: XOutOfBounds Klassenreferenz</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Übung Objektorientierte Programmierung
   &#160;<span id="projectnumber">1.0</span>
   </div>
   <div id="projectbrief">First Version 2017</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Erzeugt von Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Suchen');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Suchen');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('class_x_out_of_bounds.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#pub-methods">Öffentliche Methoden</a> &#124;
<a href="class_x_out_of_bounds-members.html">Aufstellung aller Elemente</a>  </div>
  <div class="headertitle">
<div class="title">XOutOfBounds Klassenreferenz</div>  </div>
</div><!--header-->
<div class="contents">

<p>Ausnahmeklasse Abgeleitete Klasse von std::exception. Speichert Ausnahmen und gibt diese auch wieder aus.  
 <a href="class_x_out_of_bounds.html#details">Mehr ...</a></p>

<p><code>#include &lt;<a class="el" href="_x_out_of_bounds_8hpp_source.html">XOutOfBounds.hpp</a>&gt;</code></p>
<div class="dynheader">
Klassendiagramm für XOutOfBounds:</div>
<div class="dyncontent">
<div class="center"><img src="class_x_out_of_bounds__inherit__graph.png" border="0" usemap="#_x_out_of_bounds_inherit__map" alt="Inheritance graph"/></div>
<map name="_x_out_of_bounds_inherit__map" id="_x_out_of_bounds_inherit__map">
</map>
</div>
<div class="dynheader">
Zusammengehörigkeiten von XOutOfBounds:</div>
<div class="dyncontent">
<div class="center"><img src="class_x_out_of_bounds__coll__graph.png" border="0" usemap="#_x_out_of_bounds_coll__map" alt="Collaboration graph"/></div>
<map name="_x_out_of_bounds_coll__map" id="_x_out_of_bounds_coll__map">
</map>
</div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Öffentliche Methoden</h2></td></tr>
<tr class="memitem:a9e2fc4dfa3f25d730563ff2492f40662"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_x_out_of_bounds.html#a9e2fc4dfa3f25d730563ff2492f40662">XOutOfBounds</a> (const char *msg)</td></tr>
<tr class="separator:a9e2fc4dfa3f25d730563ff2492f40662"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7655380482ddb2f03570caee39fa2748"><td class="memItemLeft" align="right" valign="top">virtual&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_x_out_of_bounds.html#a7655380482ddb2f03570caee39fa2748">~XOutOfBounds</a> ()  throw ()</td></tr>
<tr class="separator:a7655380482ddb2f03570caee39fa2748"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5a047c4fa3db7ef57de7732ae4a2e937"><td class="memItemLeft" align="right" valign="top">const char *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_x_out_of_bounds.html#a5a047c4fa3db7ef57de7732ae4a2e937">what</a> () const  throw ()</td></tr>
<tr class="memdesc:a5a047c4fa3db7ef57de7732ae4a2e937"><td class="mdescLeft">&#160;</td><td class="mdescRight">Getter Kann aufgrund von throw() selbst keine exceptions werfen.  <a href="#a5a047c4fa3db7ef57de7732ae4a2e937">Mehr ...</a><br /></td></tr>
<tr class="separator:a5a047c4fa3db7ef57de7732ae4a2e937"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Ausführliche Beschreibung</h2>
<div class="textblock"><p>Ausnahmeklasse Abgeleitete Klasse von std::exception. Speichert Ausnahmen und gibt diese auch wieder aus. </p>
</div><h2 class="groupheader">Beschreibung der Konstruktoren und Destruktoren</h2>
<a id="a9e2fc4dfa3f25d730563ff2492f40662"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a9e2fc4dfa3f25d730563ff2492f40662">&#9670;&nbsp;</a></span>XOutOfBounds()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">XOutOfBounds::XOutOfBounds </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>msg</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Konstruktor </p><dl class="params"><dt>Parameter</dt><dd>
  <table class="params">
    <tr><td class="paramname">msg</td><td>Beschreibung / Bezeichnung der Ausnahme </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<a id="a7655380482ddb2f03570caee39fa2748"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a7655380482ddb2f03570caee39fa2748">&#9670;&nbsp;</a></span>~XOutOfBounds()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">XOutOfBounds::~XOutOfBounds </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Destruktor</p>
<p>Kann aufgrund von throw() selbst keine exceptions werfen. </p>

</div>
</div>
<h2 class="groupheader">Dokumentation der Elementfunktionen</h2>
<a id="a5a047c4fa3db7ef57de7732ae4a2e937"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a5a047c4fa3db7ef57de7732ae4a2e937">&#9670;&nbsp;</a></span>what()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char * XOutOfBounds::what </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Getter Kann aufgrund von throw() selbst keine exceptions werfen. </p>
<dl class="section return"><dt>Rückgabe</dt><dd>Beschreibung / Bezeichnung der Ausnahme </dd></dl>

</div>
</div>
<hr/>Die Dokumentation für diese Klasse wurde erzeugt aufgrund der Dateien:<ul>
<li>C:/portableDevEnv_2017/workspace2017/project/src/<a class="el" href="_x_out_of_bounds_8hpp_source.html">XOutOfBounds.hpp</a></li>
<li>C:/portableDevEnv_2017/workspace2017/project/src/<a class="el" href="_x_out_of_bounds_8cpp.html">XOutOfBounds.cpp</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="class_x_out_of_bounds.html">XOutOfBounds</a></li>
    <li class="footer">Erzeugt am Sam Jun 2 2018 10:44:30 für Übung Objektorientierte Programmierung von
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
